fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/serializers/FlagFromTagDimensionToDimensionSpecSpec.groovy

```diff
@@ -0,0 +1,108 @@
+import com.yahoo.bard.webservice.data.PartialDataHandler
+import com.yahoo.bard.webservice.data.QueryBuildingTestingResources
+import com.yahoo.bard.webservice.data.config.dimension.DefaultDimensionField
+import com.yahoo.bard.webservice.data.metric.LogicalMetric
+import com.yahoo.bard.webservice.data.metric.mappers.NoOpResultSetMapper
+import com.yahoo.bard.webservice.data.volatility.DefaultingVolatileIntervalsService
+import com.yahoo.bard.webservice.druid.model.datasource.DataSource
+import com.yahoo.bard.webservice.druid.model.datasource.TableDataSource
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery
+import com.yahoo.bard.webservice.druid.model.query.GroupByQuery
+import com.yahoo.bard.webservice.table.resolver.DefaultPhysicalTableResolver
+import com.yahoo.bard.webservice.web.ApiFilter
+import com.yahoo.bard.webservice.web.DefaultFilterOperation
+import com.yahoo.bard.webservice.web.apirequest.DataApiRequest
+import com.yahoo.bard.webservice.web.filters.ApiFilters
+
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import org.joda.time.DateTime
+import org.joda.time.Hours
+import org.joda.time.Interval
+
+import spock.lang.Specification
+
+class FlagFromTagDimensionToDimensionSpecSpec extends Specification {
+
+    ObjectMapper objectMapper
+    QueryBuildingTestingResources resources
+    DruidQueryBuilder builder
+    DefaultPhysicalTableResolver resolver
+    DataApiRequest apiRequest
+    DruidAggregationQuery<?> druidQuery
+
+    def setup() {
+        objectMapper = new ObjectMappersSuite().getMapper()
+        resources = new QueryBuildingTestingResources()
+        resolver = new DefaultPhysicalTableResolver(new PartialDataHandler(), new DefaultingVolatileIntervalsService())
+        builder = new DruidQueryBuilder(
+                resources.logicalDictionary,
+                resolver,
+                resources.druidFilterBuilder,
+                resources.druidHavingBuilder
+        )
+        apiRequest = Mock(DataApiRequest)
+        LogicalMetric lm1 = new LogicalMetric(resources.simpleTemplateQuery, new NoOpResultSetMapper(), "lm1", null)
+
+        apiRequest.getTable() >> resources.lt14
+        apiRequest.getGranularity() >> HOUR.buildZonedTimeGrain(UTC)
+        apiRequest.getTimeZone() >> UTC
+        apiRequest.getLogicalMetrics() >> ([lm1])
+        apiRequest.getIntervals() >> [new Interval(new DateTime("2015"), Hours.ONE)]
+        apiRequest.getFilterDimensions() >> []
+        apiRequest.getTopN() >> OptionalInt.empty()
+        apiRequest.getSorts() >> ([])
+        apiRequest.getCount() >> OptionalInt.empty()
+        apiRequest.getApiFilters() >> new ApiFilters([(resources.d14) : [new ApiFilter(resources.d14, DefaultDimensionField.ID, DefaultFilterOperation.in, ["TRUE_VALUE"] as Collection)] as Set] as Map)
+    }
+
+    def "Flag from tag dimension based on lookup dimension serializes to cascade extraction function with tag extraction and transformation in grouping context"() {
```

Comment:
I would consider parameterizing these expected values a little more so it's easier to fix the tests is there are subtle changes.  Extract the formula for the final extraction function perhaps.